src/System.Reflection/tests/PropertyInfo/PropertyInfo_GetSetValueTests.cs

```diff
@@ -0,0 +1,374 @@
+        {
+            Object obj = Activator.CreateInstance(type);
+            PropertyInfo pi = GetProperty(type, propertyName);
+
+            Assert.Equal(pi.Name, propertyName);
+
+            Assert.Throws<TargetParameterCountException>(() =>
+            {
+                var value = pi.GetValue(obj, testObj);
+            });
+        }
+
+        [Theory]
+        [InlineData("Property1", typeof(LaterClass), null)]
+        [InlineData("PropertyC", typeof(MyCoVariantTest), new Object[] { "1", "2" })]
+        public static void TestGet_ThrowsArgumentException(string propertyName, Type type, Object[] testObj)
+        {
+            Object obj = Activator.CreateInstance(type);
+            PropertyInfo pi = GetProperty(type, propertyName);
+
+            Assert.Equal(pi.Name, propertyName);
+
+            Assert.Throws<ArgumentException>(() =>
+            {
+                var value = pi.GetValue(obj, testObj);
+            });
+        }
+
+
+        //Verify PropertyInfo.GetValue throws Exception
+        [Fact]
+        public static void TestGet_ThrowsException()
+        {
+            string propertyName = "PropertyC";
+            Object obj = Activator.CreateInstance(typeof(MyCoVariantTest));
+            PropertyInfo pi = GetProperty(typeof(MyCoVariantTest), propertyName);
+
+            Assert.Equal(pi.Name, propertyName);
+
+            // In Win8p instead of TargetException , generic Exception is thrown
+            // Refer http://msdn.microsoft.com/en-us/library/b05d59ty.aspx
+            Assert.ThrowsAny<Exception>(() => pi.GetValue(null, new Object[] { "1", "2" }));
+        }
+
+
+
+        [Theory]
+        [InlineData("NoSetterProp", 100, null)]
+        [InlineData("Item", "pw", new Object[] { 99, 2, "hello", "SOME  string" })]
+        [InlineData("Item", 100, new Object[] { 99, 2, new string[] { "hello" }, "SOME  string" })]
+        public static void TestSet_ThrowsArgumentException(string propertyName, Object setValue, Object[] index)
+        {
+            TypeWithProperties obj = new TypeWithProperties();
+            PropertyInfo pi = GetProperty(typeof(TypeWithProperties), propertyName);
+
+            Assert.Equal(pi.Name, propertyName);
+
+            Assert.Throws<ArgumentException>(() =>
+            {
+                pi.SetValue(obj, setValue, index);
+            });
+        }
+        
+        //Try to set instance Property with null object 
+        [Theory]
+        [InlineData("HasSetterProp", null, null)]
+        [InlineData("Item", "pw", new Object[] { 99, 2, "SOME string" })]
+        public static void TestSetNull_ThrowsException(string propertyName, Object setValue, Object[] index)
+        {
+            PropertyInfo pi = GetProperty(typeof(TypeWithProperties), propertyName);
+
+            Assert.Equal(pi.Name, propertyName);
+
+            // Generic Exception is thrown in WP8 instead of TargetException
+            Assert.ThrowsAny<Exception>(() => pi.SetValue(null, null, null));
```

Comment:
It's unclear to me here which null is causing the exception.  It'd be helpful to tease this apart a bit, e.g. by having an assert for each of the arguments that will throw if it's null, and by having non-null arguments for the other inputs to that call, so that we're sure that particular argument is the one causing the failure.